src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala

```diff
@@ -431,6 +431,13 @@ object macros:
           validateJsonArray(key, cursor) { value =>
             validateJsonSchema[t](key, value.hcursor)
           }
+
+        case '[Map[f, t]] =>
```

Comment:
Needed to be check in circe. Here type `f` has no bounds. I could guess that circe supports only `Map[String, t]`. Otherwise we will get unpredictable errors with a compound type `t`.
Please check this case and bound `f` to `String` if my assumption is correct.

Reply:
Well, I had similar doubts, and created a `compile error test` for Map[Int, Whatever] - and it failed with `cannot prove json structure`.
But yes, it makes perfect sense, because json keys are strings only.

Reply:
We should probably enforce String here.

Reply:
Here is a list of key types that circe supports by default.
https://github.com/circe/circe/blob/master/modules/core/shared/src/main/scala/io/circe/KeyEncoder.scala